src/optparse.coffee

```diff
@@ -25,37 +25,26 @@ exports.OptionParser = class OptionParser
   # parsers that allow you to attach callback actions for every flag. Instead,
   # you're responsible for interpreting the options object.
   parse: (args) ->
-    options = arguments: []
-    skippingArgument = no
-    originalArgs = args
-    args = normalizeArguments args
-    for arg, i in args
-      if skippingArgument
-        skippingArgument = no
-        continue
-      if arg is '--'
-        pos = originalArgs.indexOf '--'
-        options.arguments = options.arguments.concat originalArgs[(pos + 1)..]
+    state =
+      argsLeft: args[..]
+      options: {}
+    while (arg = state.argsLeft.shift())?
```

Comment:
Now that I think about it, I think we can just do a single pass over the arguments beforehand to 1. group arguments with values (`-o [dir]`) 2. explode combined options (e.g. `-abck`) 3. identify the first non-optional argument, then use a `for`. I'll do that.